### PR TITLE
Introduce DHCPv6 (i)PXE Support

### PR DESCRIPTION
**Pull request by guvenc**
Introduce DHCPv6 (i)PXE support including the test cases and RA
related refactoring and changes.

### CODE REVIEW

File: test/helpers.py [hunk 1]
```diff
@@ -51,7 +51,12 @@ def is_ipv6_tcp_pkt(pkt):
 	return IPv6 in pkt and TCP in pkt
 
 def is_tcp_pkt(pkt):
-	return TCP in pkt
+	if TCP in pkt:
+		for option in pkt[TCP].options:
+			if option[0] == 'MSS':
+				return False
+		return True
+	return False
```
Comment (PlagueCZ) [inline test/helpers.py:59]: I know it's just pytest, but failing fast would look more in line with our code.

But for a real Pythonic code:
`return TCP in pkt and 'MSS' not in [ option[0] for option in pkt[TCP].options ]`

File: test/test_dhcpv6.py [hunk 2]
```diff
@@ -2,10 +2,42 @@
...
+		ShortEnumField("optcode", 15, dhcp6opts),
+		ShortField("optlen", None),
+		PacketListField("user_class_data", [], SubOptionField,
+						length_from=lambda pkt: pkt.optlen)
+	]
+	def post_build(self, p, pay):
+		if self.optlen is None:
+			length = len(p) - 4
+			p = p[:2] + struct.pack("!H", length) + p[4:]
+		return p + pay
```
Comment (PlagueCZ) [inline test/test_dhcpv6.py:31]: I don't even know what's going on here, but I'll trust you and the test results 😄

**Review by PlagueCZ — COMMENTED** [review]
Found just some small things.
Tests pass also on my setup. 
Unable to test IPv6 on my local setup properly (VMs not ready), if Tao cannot test, I would have to update my images to try.

File: test/config.py [hunk 3]
```diff
@@ -83,6 +83,11 @@
 lb_ip6 = "2a10:defe:e01f:f4::2"
 lb_ip6_pfx = "2a10:defe:e01f:f4::2/128"
 
+#PXE realted
```
[the remark this answers is not in the document]
Comment (guvenc) [reply test/config.py:86]: :)

File: test/grpc_client.py [hunk 4]
```diff
@@ -116,6 +116,9 @@ def init(self):
 	def addinterface(self, vm_name, pci, vni, ipv4, ipv6):
 		return self._getUnderlayRoute(f"add interface --id={vm_name} --device={pci} --vni={vni} --ipv4={ipv4} --ipv6={ipv6}")
 
+	def addinterface_with_pxe(self, vm_name, pci, vni, ipv4, ipv6, pxe_server, ipxe_file):
+		return self._getUnderlayRoute(f"add interface --id={vm_name} --device={pci} --vni={vni} --ipv4={ipv4} --ipv6={ipv6} --pxe-file-name={ipxe_file} --pxe-server={pxe_server}")
+
```
[the remark this answers is not in the document]
Comment (guvenc) [reply test/grpc_client.py:121]: Oh Thanks. Suggesting the more pythonic ways of doing things. They look beautiful. I definitely need to quit writing python code as if it was c :) 
The second variant is even for me too pythonic so I will go for the first one. :)

File: test/helpers.py (hunk 1 above)
Comment (guvenc) [reply test/helpers.py:59]: definitely nicer.

File: test/test_dhcpv6.py (hunk 2 above)
Comment (guvenc) [reply test/test_dhcpv6.py:31]: Implementing the post_build from the parent class to calculate the length of this layer (if not done, already) and position it to third and fourth byte packed to network byte order.

**Review by PlagueCZ — APPROVED** [review]
Looks pretty good to me

**Pull request merged**